### PR TITLE
Bug fix: Bayer "No timestamp provided as argument” error

### PR DESCRIPTION
**Pull request by gniezen**
See https://trello.com/c/w7Nfk5yc for details.

Ping @jh-bate for code review. NOT high priority at the moment, but also just a couple of small changes ;)

### CODE REVIEW

**Comment by gniezen** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;w7Nfk5yc&#x2F;10-uploader-bug-bayer-contour-next-usb-no-timestamp-provided-as-argument"> [uploader] [bug] Bayer Contour Next USB: &quot;No timestamp provided as argument&quot;</a>

File: lib/drivers/bayerContourNext.js [hunk 1]
```diff
@@ -543,7 +545,7 @@ module.exports = function (config) {
             } else {
               recordType = result.recordType;
               // We only collect data records (R)
-              if (recordType === 'R') {
+              if (recordType === 'R' && result.timestamp) {
```
Comment (jh-bate) [inline lib/drivers/bayerContourNext.js]: there are no casing issues?

File: lib/drivers/bayerContourNext.js [hunk 2]
```diff
@@ -324,16 +324,18 @@ module.exports = function (config) {
...
                 } else {
-                  robj.timestamp = parseInt(r[4]);
-                  robj.annotations = getAnnotations(r[3], data);
-                  robj.control = isControl(r[3]);
-                  robj.units = r[2];
-                  if(robj.units === 'mmol/L') {
-                    robj.glucose = parseFloat(r[1]);
-                  } else {
-                    robj.glucose = parseInt(r[1]);
+                  if(r) {
```
Comment (jh-bate) [inline lib/drivers/bayerContourNext.js]: I assume this is declared else where but `r` isn't overly descriptive

File: lib/drivers/bayerContourNext.js [hunk 3]
```diff
@@ -324,16 +324,18 @@ module.exports = function (config) {
...
+                    robj.timestamp = parseInt(r[4]);
+                    robj.annotations = getAnnotations(r[3], data);
+                    robj.control = isControl(r[3]);
+                    robj.units = r[2];
+                    if(robj.units === 'mmol/L') {
+                      robj.glucose = parseFloat(r[1]);
+                    } else {
+                      robj.glucose = parseInt(r[1]);
```
Comment (jh-bate) [inline lib/drivers/bayerContourNext.js]: can the `parse...` functions ever produce errors? ... might not matter here but just wondering

**Comment by jh-bate** [comment]
LGTM @gniezen , just a few small comments / queries

File: lib/drivers/bayerContourNext.js (hunk 1 above)
Comment (gniezen) [inline lib/drivers/bayerContourNext.js]: @jh-bate You mean with the 'R'? No, it's specified in the spec to be a capital 'R' and should never be lowercase.

File: lib/drivers/bayerContourNext.js (hunk 2 above)
Comment (gniezen) [inline lib/drivers/bayerContourNext.js]: It's been bothering me too. I've now renamed it to `fields`.

File: lib/drivers/bayerContourNext.js (hunk 3 above)
Comment (gniezen) [inline lib/drivers/bayerContourNext.js]: @jh-bate No, they just produce NaN when they fail.

**Pull request merged**